src/ERC6538Registry.sol

```diff
@@ -24,22 +58,185 @@ contract ERC6538Registry is IERC6538Registry {
     uint256 schemeId,
     bytes memory signature,
     bytes memory stealthMetaAddress
-  ) external pure {
-    registerKeysOnBehalf(_toBytes(registrant), schemeId, signature, stealthMetaAddress);
+  ) external {
+    // Check for nonce
+    bytes32 digest = _hashTypedDataV4(
+      keccak256(
+        abi.encode(TYPE_HASH, registrant, schemeId, stealthMetaAddress, nonceOf[registrant]++)
+      )
+    );
+    require(isValidSignatureNow(registrant, digest, signature), "Invalid signature");
+    stealthMetaAddressOf[registrant][schemeId] = stealthMetaAddress;
+    emit StealthMetaAddressSet(registrant, schemeId, stealthMetaAddress);
   }
 
-  /// @inheritdoc IERC6538Registry
-  function registerKeysOnBehalf(
-    bytes memory, // registrant
-    uint256, // schemeId
-    bytes memory, // signature
-    bytes memory // stealthMetaAddress
-  ) public pure {
-    revert("not implemented");
+  /**
+   * @dev Returns the domain separator for the current chain.
+   */
```

Comment:
Nit, above the natspec is `///`, here we use the `/**`

Reply:
Updated in the same commit as above https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/4460b6edabe9487c5157a75f918ee7c116919931